pkg/nsx/services/inventory/inventory.go

```diff
@@ -0,0 +1,250 @@
+	ContainerNetworkPolicy       InventoryType = "ContainerNetworkPolicy"
+
+	// Inventory cluster type
+	InventoryClusterTypeWcp = "WCP"
+
+	InventoryClusterCniType = "NSX-Operator"
+
+	// Inventory network status
+	NetworkStatusHealthy   = "HEALTHY"
+	NetworkStatusUnhealthy = "UNHEALTHY"
+
+	// Inventory infra
+	InventoryInfraTypeVsphere = "vSphere"
+
+	InventoryMaxDisTags = 20
+	InventoryK8sPrefix  = "dis:k8s:"
+	MaxTagLen           = 256
+	MaxResourceTypeLen  = 128
+)
+
+type InventoryKey struct {
+	InventoryType InventoryType
+	ExternalId    string
+	Key           string
+}
+
+const (
+	operationCreate = "CREATE"
+	operationUpdate = "UPDATE"
+	operationDelete = "DELETE"
+	operationNone   = "NONE"
+
+	InventoryStausUp       = "UP"
+	InventoryStatusDown    = "DOWN"
+	InventoryStatusUnknown = "UNKNOWN"
+)
+
+var (
+	log         = &logger.Log
+	emptyKeySet = sets.New[InventoryKey]()
+)
+
+type InventoryService struct {
+	commonservice.Service
+	applicationInstanceStore *ApplicationInstanceStore
+	clusterStore             *ClusteStore
+
+	requestBuffer []containerinventory.ContainerInventoryObject
+	pendingAdd    map[string]interface{}
+	pendingDelete map[string]interface{}
+
+	stalePods map[string]interface{}
+}
+
+func InitializeService(service commonservice.Service) (*InventoryService, error) {
+	inventoryService := &InventoryService{
+		requestBuffer: make([]containerinventory.ContainerInventoryObject, 0),
+		pendingAdd:    make(map[string]interface{}),
+		pendingDelete: make(map[string]interface{}),
+		stalePods:     make(map[string]interface{}),
+	}
+
+	// TODO, Inventory store should have its own store
+	inventoryService.applicationInstanceStore = &ApplicationInstanceStore{ResourceStore: commonservice.ResourceStore{
+		Indexer: cache.NewIndexer(keyFunc, cache.Indexers{string(ContainerApplicationInstance): indexFunc}),
+	}}
+	inventoryService.clusterStore = &ClusteStore{ResourceStore: commonservice.ResourceStore{
+		Indexer: cache.NewIndexer(keyFunc, cache.Indexers{string(ContainerCluster): indexFunc}),
+	}}
+	inventoryService.Service = service
+	return inventoryService, nil
+}
+
+func (s *InventoryService) Init() error {
+	cluster, err := s.GetContainerCluster()
+	// If there is no such cluster, create one.
+	// Otherwise, sync with NSX for different types of inventory objects.
+	if err == nil {
+		err = s.clusterStore.Add(&cluster)
+		if err != nil {
+			log.Error(err, "Add cluster to store")
+		}
+		return err
+	}
+	log.Error(err, "Cannot find existing container cluster, will create one")
+	newContainerCluster := s.BuildInventoryCluster()
+	cluster, err = s.AddContainerCluster(newContainerCluster)
+	if err != nil {
+		return err
+	}
+	err = s.clusterStore.Add(&cluster)
+	if err != nil {
+		log.Error(err, "Add cluster to store")
+		return err
+	}
+	log.Info("A new ContainerCluster is added", "cluster", newContainerCluster.DisplayName)
+	return nil
+}
+
+func (s *InventoryService) SyncInventoryStoreByType(clusterId string) error {
```

Comment:
It looks like this function is not called?

Reply:
you're correct.  Update